### PR TITLE
IC-1406 circle ci job to manually clean and re-deploy the research environment with fresh seed data.

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

- adds a spring profile to seed the research environment with auto generated migrations
- adds a manual circleci job to clean the research database and re-deploy it, i.e. re-seed the database

## What is the intent behind these changes?

allow the research environment to be 'reset' at will.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -178,6 +189,22 @@ workflows:
           context:
             - hmpps-common-vars
             - hmpps-interventions-service-research
+      - approve_reset_research_db:
+          type: approval
+      - reset_research_db:
+          requires:
+            - approve_reset_research_db
+      - hmpps/deploy_env:
```
Comment (tomsmyers) [inline .circleci/config.yml:197]: i couldn't find a dry way to do this :(

File: .circleci/config.yml [hunk 2]
```diff
@@ -103,6 +104,16 @@ jobs:
...
           additional-arguments: << parameters.snyk-args >>
 
+  reset_research_db:
+    docker:
+      - image: 'cimg/base:stable'
+    steps:
+      - checkout
+      - hmpps/k8s_setup
+      - kubernetes/create-or-update-resource:
+          # this cleans the database - do not change this namespace!!
```
Comment (tomsmyers) [inline .circleci/config.yml:114]: this scares me

File: user_research/reset_env.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+apiVersion: batch/v1
+kind: Job
```
Comment (sldblog) [inline user_research/reset_env.yaml:2]: _not a blocker_
Given it's a kubernetes resource YAML, could we put it in `kubectl_deploy/reset_user_research_db.yaml`?

File: src/main/resources/db/research/V100_0__service_providers.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
+insert into service_provider
```
Comment (sldblog) [inline src/main/resources/db/research/V100_0__service_providers.sql:1]: _observation_
If these `V100` version numbers are causing migration problems on deployment, we can avoid them by marking them repeatable (naming it `R__something.sql`), which will not disturb the versioning and possibly doesn't require a separate deployment.

File: user_research/reset_env.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+metadata:
+  name: clean-and-migrate-user-research-env
+spec:
+  template:
+    spec:
+      securityContext:
+        runAsUser: 100
+      containers:
+        - name: clean-and-migrate-user-research-env
+          image: flyway/flyway:7.7.0-alpine
```
Comment (sldblog) [inline user_research/reset_env.yaml:12]: _observation_
It looks like if we bundle flyway into our `quay.io/hmpps/hmpps-interventions-service` image here (and maybe use repeatable migrations?), we probably won't need a separate deployment to apply the changes

**Review by sldblog — APPROVED** [review]
👍 LGTM

Just a couple of observations that may let us simplify in the future

File: user_research/reset_env.yaml (hunk 3 above)
Comment (tomsmyers) [reply user_research/reset_env.yaml:2]: sure thing happy to change this

File: src/main/resources/db/research/V100_0__service_providers.sql (hunk 4 above)
Comment (tomsmyers) [reply src/main/resources/db/research/V100_0__service_providers.sql:1]: oh interesting i'll look more into this, thanks

**Pull request merged**